core/modal_models/modes.c

```diff
@@ -557,4 +557,31 @@ void _lf_terminate_modal_reactors() {
     }
     _lf_unsused_suspended_events_head = NULL;
 }
+
+/**
+ * @brief This function accepts an array of ALL the modes ordered from top-level and ending with leaf modes, 
+ *  and adds a reactor ptr into return_vec if a mode transition was requested. This will not add reactors having 
+ *  contained modes. Only reactors that have scheduled a transition itself.
+ * 
+ * @param states 
+ * @param states_size 
+ * @param return_vec 
+ * @return int 
+ */
+int _lf_mode_collect_transitioning_reactors(reactor_mode_state_t **states, int states_size, void * _return_vec) {
+    int num_transitioning_reactors = 0;
+    vector_t *return_vec = (vector_t *) _return_vec;
```

Comment:
This function is called from code-generated functions. My thinking was to avoid including vector.h in the code-generated file to keep it out of the namespace/scope of the user reactions.